### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by reneradoi**
This PR adds the basic implementation to the repo:
- provide the rockcraft.yaml configuration file to build the rock
- add README and license files
- add basic workflows for github actions ci

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Charmed Valkey rock
```
Comment (Mehdi-Bendriss) [inline README.md:1]:
```suggestion
# Charmed Valkey rock
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+This repository will contain the packaging metadata for creating a rock of Charmed Valkey built from Charmed Valkey 
+Snap.
+
+License
```
Comment (Mehdi-Bendriss) [inline README.md:5]:
```suggestion
## License:
```

File: README.md
Comment (Mehdi-Bendriss) [inline README.md:1]: Could you detail further steps to run this rock like we have in other repos (skope, docker etc..)?

File: rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    startup: enabled
+    summary: Start Metrics Exporter
+    command: "/bin/redis_exporter"
+
+parts:
+  valkey-user:
+    plugin: nil
+    overlay-script: |
+      set -x
+      useradd -R $CRAFT_OVERLAY -M -U -r valkey
```
Comment (Mehdi-Bendriss) [inline rockcraft.yaml:36]: We should in the next "parts" perhaps "chown" files and directories to this user, right?

File: rockcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+platforms:  # The platforms this ROCK should be built on and run on
+  amd64:
+
+services:
+  valkey:
+    override: replace
+    startup: enabled
+    summary: Start Valkey
+    command: "valkey-server"
```
Comment (Mehdi-Bendriss) [inline rockcraft.yaml:24]: I believe we should ensure that this ran with the `vakey` user not root

**Review by Mehdi-Bendriss — COMMENTED** [review]
Thanks René - I left a few comments

File: README.md
Comment (reneradoi) [reply README.md:1]: done

File: rockcraft.yaml (hunk 3 above)
Comment (reneradoi) [reply rockcraft.yaml:38]: Not necessarily, valkey doesn't have specific directories it needs to own to be functional. I've now added a directory for saving appendonly-files or snapshot files to and made this the default when starting the service, just to be more explicit on this.

File: rockcraft.yaml (hunk 4 above)
Comment (reneradoi) [reply rockcraft.yaml]: done

File: rockcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+platforms:  # The platforms this ROCK should be built on and run on
+  amd64:
+
+services:
+  valkey:
+    override: replace
+    startup: enabled
+    summary: Start Valkey
+    command: "valkey-server --dir /var/lib/valkey"
+    user: valkey
```
Comment (Mehdi-Bendriss) [inline rockcraft.yaml:25]: very nice! TIL

**Review by Mehdi-Bendriss — APPROVED** [review]
Thank you! Could you please add a short description to the PR before merging?

File: rockcraft.yaml (hunk 5 above)
Comment (reneradoi) [reply rockcraft.yaml:25]: Thanks! Description was added.

**Pull request merged**